cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -443,6 +488,17 @@ struct AsciiLower {
   }
 };
 
+void TransformAsciiSwapCase(const uint8_t* input, int64_t length, uint8_t* output) {
+  std::transform(input, input + length, output, ascii_swapcase);
+}
+
+template <typename Type>
+struct AsciiSwapCase {
+  static Status Exec(KernelContext* ctx, const ExecBatch& batch, Datum* out) {
+    return StringDataTransform<Type>(ctx, batch, TransformAsciiSwapCase, out);
+  }
+};
+
```

Comment:
Add `Utf8SwapCase`

Reply:
Addressed in [ba66c91](https://github.com/apache/arrow/pull/10855/commits/ba66c919e0f4e593fa0b70b193af79ea7ec76fa4)